Common/include/linear_algebra/CSysMatrix.hpp

```diff
@@ -498,6 +498,42 @@ class CSysMatrix {
     AddBlock(block_i, block_j, val_block, OtherType(-1));
   }
 
+  /*!
+   * \brief Set the value of a scaled block in the sparse matrix.
+   * \note This is an templated overload for C2Dcontainer specialization su2matrix.
+   *       It assumes that MatrixType supports a member type Scalar and access operator[][].
+   *       If the template param Overwrite is false we add to the block (bij += alpha*b).
+   * \param[in] block_i - Row index.
+   * \param[in] block_j - Column index.
+   * \param[in] val_block - Block to set to A(i, j).
+   * \param[in] alpha - Scale factor.
+   */
+  template <class MatrixType, bool Overwrite = true>
+  inline void SetBlock(unsigned long block_i, unsigned long block_j, MatrixType& val_block,
+                       typename MatrixType::Scalar alpha = 1.0) {
+    auto mat_ij = GetBlock(block_i, block_j);
+    if (!mat_ij) return;
+    for (auto iVar = 0ul; iVar < nVar; ++iVar) {
+      for (auto jVar = 0ul; jVar < nEqn; ++jVar) {
+        *mat_ij = (Overwrite ? ScalarType(0) : *mat_ij) + PassiveAssign(alpha * val_block[iVar][jVar]);
```

Comment:
```suggestion
        *mat_ij = (Overwrite ? ScalarType(0) : *mat_ij) + PassiveAssign(alpha * val_block(iVar, jVar));
```
Safer like that to avoid depending on the storage layout of the input matrix (nice addition btw :+1:)

Reply:
I changed this for the new function.
It would be good to also make a similar change for the exisitng overloads of SetBlock. Since this led to a compiler error when tried, I suggest fixing this in a separate pull request.